bin/import_freestmandarin.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+from __future__ import absolute_import, division, print_function
+
+# Make sure we can import stuff from util/
+# This script needs to be run from the root of the DeepSpeech repository
+import os
+import sys
+sys.path.insert(1, os.path.join(sys.path[0], '..'))
+
+import argparse
+import glob
+import numpy as np
+import pandas
+import tarfile
+
+
+COLUMN_NAMES = ['wav_filename', 'wav_filesize', 'transcript']
+
+
+def extract(archive_path, target_dir):
+    print('Extracting {} into {}...'.format(archive_path, target_dir))
+    with tarfile.open(archive_path) as tar:
+        tar.extractall(target_dir)
+
+
+def preprocess_data(tgz_file, target_dir):
+    # First extract main archive and sub-archives
+    extract(tgz_file, target_dir)
+    main_folder = os.path.join(target_dir, 'ST-CMDS-20170001_1-OS')
+
+    # Folder structure is now:
+    # - ST-CMDS-20170001_1-OS/
+    #   - *.wav
+    #   - *.txt
+    #   - *.metadata
+
+    def load_set(glob_path):
+        set_files = []
+        for wav in glob.glob(glob_path):
+            wav_filename = wav
+            wav_filesize = os.path.getsize(wav)
+            txt_filename = os.path.splitext(wav_filename)[0] + '.txt'
+            with open(txt_filename, 'r') as fin:
+                transcript = fin.read()
+            set_files.append((wav_filename, wav_filesize, transcript))
+        return set_files
+
+    # Load all files, then deterministically split into train/dev/test sets
+    all_files = load_set(os.path.join(main_folder, '*.wav'))
+    df = pandas.DataFrame(data=all_files, columns=COLUMN_NAMES)
+    df.sort_values(by='wav_filename', inplace=True)
+
+    indices = np.arange(0, len(df))
+    np.random.seed(12345)
+    np.random.shuffle(indices)
+
+    test_indices = indices[-5000:]
+    dev_indices = indices[-10000:-5000]
+    train_indices = indices[:-10000]
+
```

Comment:
Why this train, dev, test split? The dev and test sizes are independent of the entire data set's size.